File: src/packets/PacketS2CZoneSource.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x0A
 * Zone Source
 * Sends a zone's current volume
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unisgned Char) Zone ID
 *      (Unsigned Char) Volume
 */
class PacketS2CZoneSource extends PacketS2C {
    constructor(ctrlrID, zoneID, sourceID) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeUInt8(sourceID);
    }

    getID() {
        return 0x0A;
    }
}

module.exports = PacketS2CZoneSource;

File: src/RNetPi.Core/Packets/PacketS2CZoneSource.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x0A
/// Zone Source
/// Sends a zone's current source
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Source ID
/// </summary>
public class PacketS2CZoneSource : PacketS2C
{
    public const byte ID = 0x0A;

    public PacketS2CZoneSource(byte controllerID, byte zoneID, byte sourceID)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        Writer.Write(sourceID);
    }

    public override byte GetID() => ID;
}